lighthouse-viewer/app/src/lighthouse-report-viewer.js

```diff
@@ -260,6 +267,23 @@ class LighthouseViewerReport {
       // noop
     }
   }
+
+  /**
+   * Initializes of a `message` listener to respond to postMessage events.
+   */
+  listenForMessages() {
+    window.addEventListener('message', e => {
+      if (e.data.lhresults) {
+        this.replaceReportHTML(e.data.lhresults);
+        ga('send', 'event', 'report', 'open in viewer');
+      }
+    });
+
+    // If the page was opened as a popup, tell the opening window we're ready.
+    if (self.opener && !self.closed) {
+      self.opener.postMessage({opened: true}, '*');
```

Comment:
Unfortunately, you can't access `opener.origin` if the opener is cross domain.

![screen shot 2016-12-20 at 1 43 18 pm](https://cloud.githubusercontent.com/assets/238208/21369146/6fda257c-c6ba-11e6-81a6-99fa6b82cc56.png)